### ISSUE TITLE
Panic from vault_pki_secret_backend_role

### ISSUE THREAD

**Issue by infa-bsurber** [hashicorp/terraform-provider-vault#429]
### Terraform Version
Terraform v0.11.14
+ provider.null v2.1.2
+ provider.vault v1.8.0

### Affected Resource(s)
- vault_pki_secret_backend_role

### Terraform Configuration Files
```hcl
resource "vault_pki_secret_backend" "intermediate_backends" {
  count = "${length(var.intermediate_names)}"

  path = "pki/${var.intermediate_names[count.index]}"

  default_lease_ttl_seconds = 31536000
  max_lease_ttl_seconds = 31536000
}

# crashes terraform
resource "vault_pki_secret_backend_role" "cert_role" {
  count = "${length(var.intermediate_names)}"

  backend = "${vault_pki_secret_backend.intermediate_backends.*.path[count.index]}"

  name = "signer"

  allow_any_name = true
  allowed_domains = "*"
  enforce_hostname = false

  ttl = 31536000
  max_ttl = 31536000
}
```

### Panic Output
https://gist.github.com/infa-bsurber/0836c5f87a5b3091261e479a07903394

### Expected Behavior
Not panicked and instead create the resource

### Actual Behavior
Panic

### Steps to Reproduce
`terraform apply`

### Important Factoids
Vault version 0.10.1
No panic on Vault version 1.1.2, so this is more of a problem of supporting older versions of Vault

**Comment by vinay-gopalan** [comment]
Thank you so much for opening this issue, and we apologies for the delay! This issue wasn't reproducible in the latest releases of Vault and TFVP, and is confirmed to be resolved in versions past TFVP v1.8.0 and Vault 1.1.2. Please let us know if you experience any further issues with the resource!

**Comment by evilmog** [comment]
I know this is an old issue but it popped up again

panic: unknown resource type vault_pki_secret_backend

goroutine 603 [running]:
github.com/hashicorp/terraform/plugin.(*GRPCProvider).getResourceSchema(0xe111a5d?, {0xc001e10cf0, 0x18})
	github.com/hashicorp/terraform@v0.12.31/plugin/grpc_provider.go:92 +0x10b
github.com/hashicorp/terraform/plugin.(*GRPCProvider).ReadResource(0xc000a17650, {{0xc001e10cf0, 0x18}, {{{0xf0b0bd8, 0xc000a39f80}}, {0xcb81b00, 0xc00055f7d0}}, {0x152ef6d8, 0x0, 0x0}})
	github.com/hashicorp/terraform@v0.12.31/plugin/grpc_provider.go:321 +0x74
github.com/GoogleCloudPlatform/terraformer/terraformutils/providerwrapper.(*ProviderWrapper).Refresh(0xc000a162a0, 0xc001e2c320, 0xc001e2c2d0)
	github.com/GoogleCloudPlatform/terraformer/terraformutils/providerwrapper/provider.go:171 +0x3a9
github.com/GoogleCloudPlatform/terraformer/terraformutils.(*Resource).Refresh(0xc001fdc990, 0x26?)
	github.com/GoogleCloudPlatform/terraformer/terraformutils/resource.go:130 +0x4e
github.com/GoogleCloudPlatform/terraformer/terraformutils.RefreshResourceWorker(0x0?, 0x0?, 0x0?)
	github.com/GoogleCloudPlatform/terraformer/terraformutils/utils.go:153 +0xad
created by github.com/GoogleCloudPlatform/terraformer/terraformutils.RefreshResources
	github.com/GoogleCloudPlatform/terraformer/terraformutils/utils.go:80 +0xe7
	
terraform 1.3.5

Provider block below

terraform {
	required_providers {
		vault = {
	    version = "~> 3.11.0"
		}
  }
}

Vault version 1.12.1
